packages/next/next-server/server/next-server.ts

```diff
@@ -196,7 +196,12 @@ export default class Server {
     const pagesManifestPath = join(this.serverBuildDir, PAGES_MANIFEST)
 
     if (!dev) {
-      this.pagesManifest = require(pagesManifestPath)
+      this.pagesManifest = require(pagesManifestPath) as PagesManifest
+      if (this.pagesManifest.nextVersion !== process.env.__NEXT_VERSION) {
+        throw new Error(
```

Comment:
I think it'd be better to do a warning instead to make sure this doesn't break existing users

Reply:
Done! 8c764ec